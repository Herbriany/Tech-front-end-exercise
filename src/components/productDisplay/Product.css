.header--products {
  font-size: 2rem;
}

.product {
  margin-bottom: 20px;
  padding: 20px;
  border: 1px solid #6C757D;
  background-color: #FFF;
  height: 340px;
  border-radius: 5px;
}

.product__information {
  height: 220px;
}
/*# sourceMappingURL=Product.css.map */