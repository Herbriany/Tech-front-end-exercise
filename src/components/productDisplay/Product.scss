@import "../../assets/scss/_variables";

.header--products {
    font-size: 2rem;
}

.product {
    margin-bottom: 20px;
    padding: 20px;
    border: 1px solid $grey-border;
    background-color: #FFF;
    height: 340px;
    border-radius: 5px;

    &__information {
        height: 220px;
    }
}