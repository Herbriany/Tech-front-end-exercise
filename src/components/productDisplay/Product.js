import React from 'react'
import PropTypes from 'prop-types'
import './Product.css'

function Product({ product, onAddProductClick }) {
    return (
        <div className="col-lg-3 col-sm-12 col-md-12">
            <div className="container product">
                <div className="product__information">
                    <img alt={product.title} src={product.image} />
                    <p>{product.title}, {product.sku}</p>
                </div>
                <p>&pound;{product.price}</p>
                <button type="button" name={product.title} className="btn btn-outline-secondary" aria-label={"Add" + product.title + "to basket. Priced " + product.price} onClick={() => onAddProductClick(product)}>Add to Basket</button>
            </div>
        </div>
    )
}

Product.propTypes = {
    product: PropTypes.object.isRequired,
    onAddProductClick: PropTypes.func.isRequired
}

export default Product
