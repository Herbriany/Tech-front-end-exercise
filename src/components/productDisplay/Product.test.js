import React from 'react'
import renderer from 'react-test-renderer'
import { products } from '../../../tools/mockData'
import { shallow } from 'enzyme'
import Product from './Product'

function render(args) {
    const defaultProps = {
        product: products[0],
        onAddProductClick: jest.fn()
    }
    const props = {...defaultProps, ...args}
    return shallow(<Product {...props}/>)
}

it("renders single product",() => {
    const tree = renderer.create(<Product product={products[0]} onAddProductClick={() =>{}} />)
    expect(tree).toMatchSnapshot()
})

it("should be a button saying Add to Basket", () => {
 const wrapper = render({})
 expect(wrapper.find("button").text()).toEqual("Add to Basket")
})
