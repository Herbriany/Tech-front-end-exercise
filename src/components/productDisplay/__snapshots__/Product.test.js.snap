// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders single product 1`] = `
<div
  className="col-lg-3 col-sm-12 col-md-12"
>
  <div
    className="container product"
  >
    <div
      className="product__information"
    >
      <img
        alt="Russell Hobbs Pennine Illuminating S Steel Kettle 20444"
        src="https://media.4rgos.it/s/Argos/9805244_R_SET?w=110&h=130"
      />
      <p>
        Russell Hobbs Pennine Illuminating S Steel Kettle 20444
        , 
        549/3179
      </p>
    </div>
    <p>
      Â£
      24.99
    </p>
    <button
      aria-label="AddRussell Hobbs Pennine Illuminating S Steel Kettle 20444to basket. Priced 24.99"
      className="btn btn-outline-secondary"
      name="Russell Hobbs Pennine Illuminating S Steel Kettle 20444"
      onClick={[Function]}
      type="button"
    >
      Add to Basket
    </button>
  </div>
</div>
`;
